core/store/src/trie/trie_storage.rs

```diff
@@ -1,33 +1,200 @@
 use std::borrow::Borrow;
-use std::collections::{HashMap, HashSet};
+use std::collections::{HashMap, HashSet, VecDeque};
 use std::sync::{Arc, Mutex};
 
 use near_primitives::hash::CryptoHash;
 
 use crate::db::refcount::decode_value_with_rc;
 use crate::trie::POISONED_LOCK_ERR;
-use crate::{DBCol, StorageError, Store};
+use crate::{metrics, DBCol, StorageError, Store};
 use lru::LruCache;
 use near_primitives::shard_layout::ShardUId;
 use near_primitives::types::{TrieCacheMode, TrieNodesCount};
 use std::cell::{Cell, RefCell};
 use std::io::ErrorKind;
 
-/// Wrapper over LruCache which doesn't hold too large elements.
+pub(crate) struct BoundedQueue<T> {
```

Comment:
When https://github.com/jeromefroe/lru-rs/pull/152 gets in, we should be a able to completely remove this `BoundedQueue`. I think it is equivalent to just putting the freshly deleted items to the end of LRU. 


:thinking: actually, I think it isn't quite equivalent. It seems that we also want so use this queue to actually protect from deletions, no? The size of the queue is large, `100_000`, I think the intention there is to keep deleted keys in-cache for at least three blocks

>Now we delay removal by 100_000, which helps to keep all nodes from 3 completely full last blocks.

I think this isn't quite correct though, there's no guarantee that deleted nodes will survive three blocks. This only works if there *are* 100_000 deleted nodes. If we delete just, say, three nodes when processing fork1, we'll most likely evict those nodes when processing fork2, as those three nodes would be the first candidates for eviction. 

Are we ok with this behavior?